pkg/bulletproof/range_batch_prover.go

```diff
@@ -0,0 +1,348 @@
+package bulletproof
+
+import (
+	crand "crypto/rand"
+
+	"github.com/gtank/merlin"
+	"github.com/pkg/errors"
+
+	"github.com/coinbase/kryptology/pkg/core/curves"
+)
+
+// BatchProve proves that a list of scalars v are in the range n.
+// It implements the aggregating logarithmic proofs defined on pg21.
+// Instead of taking a single value and a single blinding factor, BatchProve takes in a list of values and list of
+// blinding factors.
+func (prover *RangeProver) BatchProve(v, gamma []curves.Scalar, n int, proofGenerators RangeProofGenerators, transcript *merlin.Transcript) (*RangeProof, error) {
+	// Define nm as the total bits required for secrets, calculated as number of secrets * n
+	m := len(v)
+	nm := n * m
+	// nm must be less than the number of generators generated
+	if nm > len(prover.generators.G) {
```

Comment:
Shouldn’t this be `nm >= len(…)`?

Reply:
ah, the code is correct, the comment isn't. nm just has to be less than or equal to the number of generators. Updated the comment/ error msg. Corrected in range prover as well.